staging/src/k8s.io/client-go/util/jsonpath/jsonpath_test.go

```diff
@@ -284,7 +358,7 @@ func TestStructInput(t *testing.T) {
 	failStoreTests := []jsonpathTest{
 		{"invalid identifier", "{hello}", storeData, "unrecognized identifier hello", false},
 		{"nonexistent field", "{.hello}", storeData, "hello is not found", false},
-		{"invalid array", "{.Labels[0]}", storeData, "map[string]int is not array or slice", false},
+		{"invalid object", "{.Labels[0]}", storeData, "invalid jsonpath for objects. Index should be an asterisk when applied over an object", false},
```

Comment:
if we're adding []-notation support for objects (which appears valid according to https://goessner.net/articles/JsonPath/index.html#e2) then I'd expect this to behave like a not-found key

Reply:
Yeah we can have that. But going through docs and online jsonpath evaluators, the result I get is `invalid` since its the violation of **jsonpath syntax**. hence, thought of keeping as an error than a `not found` based successful response.

Reply:
the thing is, labels is a map. index of 0 in map for jsonpath is invalid.

Reply:
that's fair... in that case, I'd expect .Labels[0] to complain about an index being used for a map, and .Labels['0'] to be handled however a "key not found" case is handled

Reply:
sure. How about changing the error message to `key not found: used index for a map`. Should we go with this?